packages/query-core/src/query.ts

```diff
@@ -452,7 +452,13 @@ export class Query<
       abort: abortController?.abort.bind(abortController),
       onSuccess: (data) => {
         if (typeof data === 'undefined') {
-          onError(new Error('Query data cannot be undefined') as any)
+          onError(
+            new Error(
+              `Query data cannot be undefined - affected query key: ${JSON.stringify(
+                this.queryKey,
+              )}`,
+            ) as any,
```

Comment:
awesome, thank you. What we should likely use instead of stringifying the querykey is just `this.queryHash`, as this contains the stringified version already, and also respects custom hash functions passed in by the user.

Reply:
Awesome, I wasn't aware of that one!